code/modules/mob/living/carbon/xenomorph/castes/slime/abilities_slime.dm

```diff
@@ -0,0 +1,480 @@
+	if(attached_atom)
+		end_ability()
+	xeno_owner.flags_pass = PASSTABLE|PASSFIRE
+	addtimer(CALLBACK(xeno_owner, TYPE_PROC_REF(/mob/living/carbon/xenomorph, reset_flags_pass)), 6)
+	RegisterSignal(owner, COMSIG_MOVABLE_IMPACT, PROC_REF(atom_hit))
+	RegisterSignal(owner, COMSIG_MOVABLE_POST_THROW, PROC_REF(throw_complete))
+	xeno_owner.throw_at(A, SLIME_POUNCE_RANGE, SLIME_POUNCE_SPEED, xeno_owner)
+	succeed_activate()
+	add_cooldown()
+	return TRUE
+
+/// Called whenever the Slime hits an atom.
+/// Will attempt to "attach" to it (buckle on mobs, place inside the atom otherwise).
+/datum/action/xeno_action/activable/slime_pounce/proc/atom_hit(datum/source, atom/hit_atom)
+	throw_complete()
+	attached_atom = hit_atom
+	owner.balloon_alert_to_viewers("Attached")
+	owner.layer = ABOVE_MOB_LAYER
+	RegisterSignal(owner, list(
+		COMSIG_ATOM_EX_ACT, // Hit by an explosion.
+		COMSIG_MOB_DEATH, // Slime died.
+		COMSIG_XENOMORPH_FIRE_BURNING, // Slime is burning.
+		COMSIG_LIVING_DO_RESIST, // Slime used the Resist action.
+		COMSIG_LIVING_STATUS_STUN, // Slime was stunned.
+		COMSIG_LIVING_STATUS_KNOCKDOWN, // Slime was knocked down.
+		COMSIG_LIVING_STATUS_PARALYZE, // Slime was paralyzed.
+		COMSIG_LIVING_STATUS_UNCONSCIOUS, // Slime was unconscious.
+		COMSIG_LIVING_STATUS_SLEEP, // Slime was asleep.
+		COMSIG_LIVING_STATUS_STAGGER), // Slime was staggered.
+		PROC_REF(call_end_ability))
+	RegisterSignal(attached_atom, COMSIG_PARENT_QDELETING, PROC_REF(call_end_ability)) // Attached atom was qdeleted.
+	RegisterSignal(attached_atom, COMSIG_ATOM_EX_ACT, PROC_REF(call_attached_atom_ex_act)) // Attached atom was hit by an explosion.
+	RegisterSignal(attached_atom, COMSIG_ATOM_BULLET_ACT, PROC_REF(call_attached_atom_projectile_hit)) // Attached atom was hit by a projectile.
+	if(isliving(attached_atom))
+		var/mob/living/attached_living = attached_atom
+		attached_living.buckle_mob(owner, TRUE, TRUE, 90)
+		attached_living.add_movespeed_modifier(MOVESPEED_ID_SLIME_ATTACHED, TRUE, 0, NONE, TRUE, SLIME_POUNCE_ATTACHED_SLOWDOWN) // An attached Slime will weigh down its victim.
+		RegisterSignal(attached_living, list(
+			COMSIG_MOVABLE_UNBUCKLE, // Slime was unbuckled.
+			COMSIG_MOB_DEATH, // Attached victim died.
+			COMSIG_LIVING_STATUS_UNCONSCIOUS), // Attached victim was unconscious.
+			PROC_REF(call_end_ability))
+		RegisterSignal(attached_living, COMSIG_LIVING_DO_RESIST, PROC_REF(call_living_resist)) // Attached victim used the Resist action.
+		if(passive_pounce)
+			return
+		var/mob/living/carbon/xenomorph/xeno_owner = owner
+		if(xeno_owner.issamexenohive(attached_living) || HAS_TRAIT(attached_living, TRAIT_INTOXICATION_IMMUNE))
+			return
+		effect_over_time(SLIME_POUNCE_INTOXICATION_STACKS + xeno_owner.xeno_caste.additional_stacks) // Effects applied every few seconds to a Slime's victim.
+		return
+	previous_turf = get_turf(owner)
+	ADD_TRAIT(owner, TRAIT_IMMOBILE, TRAIT_GENERIC)
+	owner.forceMove(hit_atom)
+	owner.anchored = TRUE
```

Comment:
Dont you want this to only handle turfs?
For objects you can just buckle normally

Reply:
Yes. Yes I do, Slimes probably shouldn't be entering vendors.

Having issues with getting mobs buckled to objects, will take a look at that later.